roles/st2web/tasks/certificate.yml

```diff
@@ -0,0 +1,37 @@
+---
+- name: Verify if custom SSL certificate was correctly specified
+  fail:
+    msg: "When using custom certificate, both 'st2web_ssl_certificate' and 'st2web_ssl_certificate_key' must be provided"
+  # no XOR in Yaml
+  when: (st2web_ssl_certificate and not st2web_ssl_certificate_key) or (not st2web_ssl_certificate and st2web_ssl_certificate_key)
+
+- name: Create SSL certificate directory
+  become: yes
+  file:
+    state: directory
+    dest: /etc/ssl/st2
+    mode: 0700
+
+- name: Save custom SSL certificate
+  become: yes
+  copy:
+    content: "{{ item.cert }}"
+    dest: "{{ item.path }}"
+    mode: 0600
```

Comment:
Makes sense.
Now wondering what user:group should it be.

Hmm, `root:root` or something different, example: https://superuser.com/a/556496